packages/core/app/src/wallets.ts

```diff
@@ -0,0 +1,172 @@
+import type {
+    WindowAppReadyEvent as WindowAppReadyEventInterface,
+    WindowAppReadyEventAPI,
+    DEPRECATED_WalletsCallback,
+    DEPRECATED_WalletsWindow,
+    Wallet,
+    WalletEventsWindow,
+} from '@wallet-standard/base';
+
+/** TODO: docs */
+export interface Wallets {
+    /**
+     * TODO: docs
+     */
+    register(...wallets: Wallet[]): () => void;
+
+    /**
+     * TODO: docs
+     */
+    get(): ReadonlyArray<Wallet>;
+
+    /**
+     * TODO: docs
+     */
+    on<E extends WalletsEventNames = WalletsEventNames>(event: E, listener: WalletsEvents[E]): () => void;
+}
+
+/** TODO: docs */
+export interface WalletsEvents {
+    /**
+     * Emitted when wallets are registered.
+     *
+     * @param wallets Wallets that were registered.
+     */
+    register(...wallets: Wallet[]): void;
+
+    /**
+     * Emitted when wallets are unregistered.
+     *
+     * @param wallets Wallets that were unregistered.
+     */
+    unregister(...wallets: Wallet[]): void;
+}
+
+/** TODO: docs */
+export type WalletsEventNames = keyof WalletsEvents;
+
+let wallets: Wallets | undefined = undefined;
+const registered = new Set<Wallet>();
+const listeners: { [E in WalletsEventNames]?: WalletsEvents[E][] } = {};
+
+/**
+ * TODO: docs
+ */
+export function getWallets(): Wallets {
+    if (wallets) return wallets;
+    wallets = Object.freeze({ register, get, on });
+
+    if (typeof window === 'undefined') return wallets;
+    const w = window as WalletEventsWindow;
+    const api = Object.freeze({ register });
+
+    try {
+        w.addEventListener('wallet-standard:register-wallet', ({ detail: callback }) => callback(api));
```

Comment:
I thought about this but it's unclear where to put them exactly -- the `app` and `wallet` packages are designed to be used by completely separate consumers, and there's no higher level package for them to go in (the `base` package having no implementation).

Reply:
I opted not to export these for now, since we really want apps and wallets to use the exported functions we provide for setting this up, and because both the app and wallet packages need both strings.